### ISSUE TITLE
gt_classes in VOC2012.py

### ISSUE THREAD

**Issue by SchrodingerLLX** [showlab/DatasetDM#17]
你好，有个问题想请教一下，在VOC2012.py中gt_classes为什么是1而不是select_class？

instances = {}
mapper_classes = [1]
instances["gt_classes"] = torch.tensor(mapper_classes, dtype=torch.int64)
masks = []
masks.append(mask == 1)

https://github.com/showlab/DatasetDM/blob/7545a68f35d8aad37f7963c326ae7c759a9a89d2/dataset/VOC2012.py#L233C2-L234C82 #

**Comment by Z1zs** [comment]
I have the same question, and I'm even more confused cause the class prediction of instance segmentation task is all "bicycle"(class 1). Have you figure it out?

**Comment by qwe3535** [comment]
I have the same question, and I'm even more confused cause the class prediction of instance segmentation task is all "bicycle"(class 1). Have you figure it out?

**Comment by Z1zs** [comment]
Hi, you can find the reply from the author here:
https://github.com/showlab/DatasetDM/issues/15#issuecomment-1791908352

To summarize, this work can only be applied in category-agnostic instance segmentation settings. You may want to try similar work like Dataset Diffusion or DiffMask.

**Comment by qwe3535** [comment]
> Hi, you can find the reply from the author here: [#15 (comment)](https://github.com/showlab/DatasetDM/issues/15#issuecomment-1791908352)
> 
> To summarize, this work can only be applied in category-agnostic instance segmentation settings. You may want to try similar work like Dataset Diffusion or DiffMask.








Okay, thank you very much for your reply, it has been very helpful to me.

May I ask if you have reproduced the performance of the paper on semantic segmentation on VOC2012?

Because there are some issues with the code, I have not been able to reproduce the performance of the paper on semantic segmentation.

In VOC2012.py,
since mapper_classes = [1]
and dataset_dict["classes_str"] = [self.classes[el] for el in mapper_classes],
this means dataset_dict["classes_str"] is only [aeroplane].

In the code for generating semantic segmentation labels in train_semantic_voc.py,

outputs = seg_model(diffusion_features, controller, prompts, tokenizer, text_embeddings)；
text_embedding is generated from class_name,
class_name is dataset_dict["classes_str"],
which means class_name is [aeroplane]. Is this correct?

Why is text_embedding generated from batch["prompt"]?
